presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsNodePartitioningProvider.java

```diff
@@ -0,0 +1,86 @@
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorSplit;
+import com.facebook.presto.spi.Node;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.connector.ConnectorNodePartitioningProvider;
+import com.facebook.presto.spi.connector.ConnectorPartitioningHandle;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.ToIntFunction;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.tpcds.Types.checkType;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+
+public class TpcdsNodePartitioningProvider
+        implements ConnectorNodePartitioningProvider
+{
+    private final String connectorId;
+    private final NodeManager nodeManager;
+    private final int splitsPerNode;
+
+    public TpcdsNodePartitioningProvider(String connectorId, NodeManager nodeManager, int splitsPerNode)
+    {
+        checkArgument(splitsPerNode > 0, "splitsPerNode must be at least 1");
+
+        this.connectorId = connectorId;
+        this.nodeManager = nodeManager;
+        this.splitsPerNode = splitsPerNode;
+    }
+
+    @Override
+    public Map<Integer, Node> getBucketToNode(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorPartitioningHandle partitioningHandle)
+    {
+        Set<Node> nodes = nodeManager.getActiveDatasourceNodes(connectorId);
+        checkState(!nodes.isEmpty(), "No TPCDS nodes available");
+
+        // Split the data using split and skew by the number of nodes available.
+        ImmutableMap.Builder<Integer, Node> bucketToNode = ImmutableMap.builder();
+        int partNumber = 0;
+        for (Node node : nodes) {
+            for (int i = 0; i < splitsPerNode; i++) {
+                bucketToNode.put(partNumber, node);
+                partNumber++;
+            }
+        }
+        return bucketToNode.build();
+    }
+
+    @Override
+    public ToIntFunction<ConnectorSplit> getSplitBucketFunction(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorPartitioningHandle partitioningHandle)
+    {
+        return value -> checkType(value, TpcdsSplit.class, "value").getPartNumber();
+    }
+
+    @Override
+    public BucketFunction getBucketFunction(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorPartitioningHandle partitioningHandle, List<Type> partitionChannelTypes, int bucketCount)
+    {
+        long totalRows = checkType(partitioningHandle, TpcdsPartitioningHandle.class, "functionHandle").getTotalRows();
+        long rowsPerBucket = totalRows / bucketCount;
```

Comment:
How you handle reminder?
E.g. 10 rows in 3 buckets. 1 remaining won't be assigned